### PR TITLE
Don't use subselect in `DeleteIssuesByRepoID`

### PR DESCRIPTION
**Pull request by JakobDev**
Part of https://codeberg.org/forgejo/discussions/issues/61

This is workaround for a bug in MariaDB

### CODE REVIEW

File: models/issues/issue_update.go [hunk 1]
```diff
@@ -696,85 +696,100 @@ func UpdateReactionsMigrationsByType(gitServiceType api.GitServiceType, original
...
-	}
+	for {
+		issueIDs := make([]int64, 0, db.DefaultMaxInSize)
 
-	// Dependencies for issues in this repository
-	if _, err = sess.In("issue_id", deleteCond).
-		Delete(&IssueDependency{}); err != nil {
-		return nil, err
-	}
+		err := sess.Table(&Issue{}).Where("repo_id = ?", repoID).Limit(db.DefaultMaxInSize).Cols("id").Find(&issueIDs)
```
Comment (lunny) [inline models/issues/issue_update.go]: We need an explicit `OrderBy` here.

File: models/issues/issue_update.go [hunk 2]
```diff
@@ -696,85 +696,100 @@ func UpdateReactionsMigrationsByType(gitServiceType api.GitServiceType, original
...
+		if err != nil {
+			return nil, err
+		}
 
-	if _, err = db.DeleteByBean(ctx, &Issue{RepoID: repoID}); err != nil {
-		return nil, err
+		_, err = sess.In("dependent_issue_id", issueIDs).Delete(&Comment{})
+		if err != nil {
+			return nil, err
+		}
```
Comment (delvh) [inline models/issues/issue_update.go:761]: Can't we shorten this to
```go
for _, issueID := range issueIDs {
  db.DeleteByBean(&TrackedTime{IssueID: issueID}, &Comment{DependentIssueID: issueID}, …)
}
```
?

Comment (delvh) [reply models/issues/issue_update.go:761]: Ideally, we even query the attachments beforehand, then we can even put them and the issues themselves inside this statement.

File: models/issues/issue_update.go (hunk 1 above)
Comment (delvh) [reply models/issues/issue_update.go]: Why?

Comment (lunny) [reply models/issues/issue_update.go]: To keep the query order explicitly and not affected by database global setting.

Comment (delvh) [reply models/issues/issue_update.go]: Ah, I overlooked the endless `for` loop.
That one wouldn't be needed either anymore with my approach.

File: models/issues/issue_update.go (hunk 2 above)
Comment (JakobDev) [reply models/issues/issue_update.go:761]: Why should be run e.g for a repo with 100 issues the delete statement 100 times instead of once? This will hurt the performance.

**Review by lunny — APPROVED** [review]

**Review by lunny — APPROVED** [review]

**Review by puni9869 — APPROVED** [review]
